common/helpers/src/avro.rs

```diff
@@ -0,0 +1,235 @@
+/// let serialized_schema = fingerprint_raw_schema(raw_schema);
+/// assert!(serialized_schema.is_ok());
+/// let schema = schema_result.unwrap().1;
+/// let translated_schema = translate_schemas(&vec![schema]);
+/// assert!(translated_schema.is_ok());
+/// ```
+pub fn translate_schemas(serialized_schemas: Vec<Vec<u8>>) -> Result<Vec<Schema>, AvroError> {
+	let mut schemas = Vec::new();
+	for s in serialized_schemas {
+		schemas.push(translate_schema(s)?);
+	}
+	Ok(schemas)
+}
+
+/// Function to get the schema writer with default container as Vec<u8>
+/// # Examples
+/// ```
+/// use avro::*;
+/// let writer = get_schema_data_writer();
+/// ```
+pub fn get_schema_data_writer<'a>(schema: &'a Schema) -> Writer<'a, Vec<u8>> {
+	Writer::with_codec(schema, Vec::new(), Codec::Snappy)
+}
+
+/// Function to populate a given schema writer with schema data.
+/// # Arguments
+/// * `writer` - io Writer ( can be obtained from get_schema_data_writer() )
+/// * `records` - record to be written as HashMap<String, SchemaValue> where SchemaValue is common types
+/// # Examples
+/// ```
+/// use avro::*;
+/// let writer = get_schema_data_writer();
+/// let record = vec![("name".to_string(), SchemaValue::String("John".to_string())), ("favorite_number".to_string(), SchemaValue::Int(42))];
+/// populate_records(writer, record);
+/// ```
+pub fn populate_records(
+	writer: &mut Writer<Vec<u8>>,
+	records: &HashMap<String, SchemaValue>,
+) -> Result<(), AvroError> {
+	let mut record_list = Record::new(writer.schema()).unwrap();
+	for (field_name, field_value) in records.iter() {
+		record_list.put(field_name, field_value.clone());
+	}
+	writer.append(record_list)?;
+	Ok(())
+}
+
+/// Function to populate a given schema with data and return serialized record.
+/// # Arguments
+/// * `schema` - Avro schema
+/// * `record` - record to be written as HashMap<String, SchemaValue> where SchemaValue is common types
+/// # Examples
+/// ```
+/// use avro::*;
+/// let schema = r#"{"type": "record", "name": "User", "fields": [{"name": "name", "type": "string"}, {"name": "favorite_number", "type": "int"}]}"#;
+/// let record = vec![("name".to_string(), SchemaValue::String("John".to_string())), ("favorite_number".to_string(), SchemaValue::Int(42))];
+/// let schema_fingerprint = fingerprint_raw_schema(schema);
+/// assert!(schema_fingerprint.is_ok());
+/// let serialized_record = populate_schema_and_serialize(schema_fingerprint.unwrap().1, record);
+pub fn populate_schema_and_serialize(
+	schema: &Schema,
+	record: &HashMap<String, SchemaValue>,
+) -> Result<Vec<u8>, AvroError> {
+	let mut writer = Writer::with_codec(schema, Vec::new(), Codec::Snappy);
+	let mut record_list = Record::new(writer.schema()).unwrap();
+	for (field_name, field_value) in record.iter() {
+		record_list.put(field_name, field_value.clone());
+	}
+	writer.append(record_list)?;
```

Comment:
So one of the main reasons to use a binary serialization scheme like `avro` was to compress the stored message.
This library as far as I've checked always writes header (which includes schema) when serializing. This actually increases the size of each individual message instead of decreasing it. We need a way to avoid serializing the schema on each write.

https://github.com/apache/avro/blob/master/lang/rust/avro/src/writer.rs#L94
https://github.com/apache/avro/blob/master/lang/rust/avro/src/writer.rs#L325

Reply:
@aramikm 
https://github.com/apache/avro/blob/75e74ee45c6a6f250a8f1321537e0fbb6c97517d/lang/rust/avro/src/writer.rs#L94

by default has_header is false i suppose and hence it will skip writing header by default

Reply:
I think you are right, by default it sets the header, and looks like setter on has_header is set to skip

Reply:
@aramikm  this looks like does our job , what you think

https://github.com/apache/avro/blob/75e74ee45c6a6f250a8f1321537e0fbb6c97517d/lang/rust/avro/src/writer.rs#L382

Reply:
@aramikm  i think this is the answer https://github.com/apache/avro/blob/75e74ee45c6a6f250a8f1321537e0fbb6c97517d/lang/rust/avro/src/writer.rs#L382

We can safely use it for internal use, what you think?

Reply:
@aramikm  I found a way, there is a lesser known function to_avro_datum and from_avro_datum that will not write to header and since we are using for internal purposes we can enforce getter of data via this process, check the latest update to function ```get_schema_data_map```
Which is only data getter from a serialized_data and its schema

Reply:
This would be great. Now the only thing that I'm curious about it how smaller is a data serialized via headerless avro vs it's json but we can do a check later.